lib/types.nix

```diff
@@ -583,6 +583,9 @@ rec {
       functor = (defaultFunctor name) // { wrapped = [ t1 t2 ]; };
     };
 
+    oneOrMore = t: either t (listOf t)
```

Comment:
Data point: I expected oneOrMore to be 'listOf t' with minimum list length of 1, not two "separate" types.

Reply:
Well, you are right but in the context of a subset of Nix mapping to INI or TOML etc. it makes a difference. For example, if the top-level of an attrset is a `either t or (listOf t)`, this would be valid:
```nix
{ 
  database.address = "127.0.0.1";
  database.ports = [ 8001 ]; 
}
```
and would turn into something like
```TOML
[database]
address = "127.0.0.1"
ports = [ 8001 ]
```
On the other hand, `database.address = "127.0.0.1"` would not typecheck because it's not a list.

Maybe the name `oneOrMore` is not appropriate, because it appears to be a haskell-like `NonEmpty` but it's actually a different sum-type.

Reply:
Yeah, I would not expect a type called `oneOrMore` to allow empty list as a value.

As is, the type is just a regular `listOf` which accepts omitting two characters (`[` and `]`) when passing a singleton. I do not see much value in that.

Reply:
Agreed, it's not really that useful.